### ISSUE TITLE
Rebalance of exotic/dark matter production

### ISSUE THREAD

**Issue by angarg12** [angarg12/incremental_table_elements#104]
Right now the game uses a simple logarithm to calculate the production of exotic/dark matter on prestige.

The problem is that the production goes up rapidly and then drops off sharply. Therefore, the best way to progress is to make short loops and idling hardly matters.

We would like to make the line 'flatter and lower' so that the amount of resources initially obtained is lower, but idling longer time pays off.

We need to do a proper analysis of the current system and how we can tweak it accordingly. Ideally production should increase almost linearly for about a day and then drop off.

**Comment by HeinousTugboat** [comment]
So, I spent some time writing a little script that runs through a full 24 hours of simulation, and gave it a standard semi-optimal strategy of just buying whichever generator is the most efficient for the next purchase. Here's a graph of the total current rate by second over 24 hours on a log-scale.

![image](https://user-images.githubusercontent.com/15902166/29998943-5831e620-9005-11e7-90b8-424b5bed05c0.png)

I haven't added in the upgrades, that's my next goal. Are you wanting to widen the climb out so it lasts closer to the whole 24 hours?

**Comment by angarg12** [comment]
That is really cool! Yes, upgrades should be included. They will become particularly relevant when more are added.

My current line of though is the following: That plot shows the net gains, and exotic matter is just the plain log of that.

If you take the log, you would get about 16 xMatter after a few minutes, about 18 after 2 hours and about 20 after 10 hours, and then gets flat. 

That progression is too draconian. Idling doesn't pay off and the best way to progress is fast prestiges. I would like for idling to pay off in an almost linear fashion, at least until about 24 hours.

I was looking at [http://blog.kongregate.com/the-math-of-idle-games-part-iii/](http://blog.kongregate.com/the-math-of-idle-games-part-iii/) to get inspiration, and a lot of games use sqrt instead of log. I checked both formulas online and indeed sqrt seems to give a better progression curve than log. Or both can be combined together.

![desmos-graph](https://user-images.githubusercontent.com/5389189/30001808-6ff2f3dc-9099-11e7-84b0-d1d903f766ec.png)
[link](https://www.desmos.com/calculator/admckyl6mw)

Red is log, blue is sqrt and green is both added together.

BTW we could use the both to plot together different prestige runs and see how they stack together, to balance the generators power/price (they will need to be toned down since the new upgrades will make them much more powerful).

**Comment by HeinousTugboat** [comment]
Hokay, here's the same chart with upgrades purchased with the same formula. Had to rewrite a huge chunk of my script to process the upgrades alongside the generators. I went ahead and [put it up on my branch](https://github.com/HeinousTugboat/nucleogenesis/blob/hind-dev/analysis/sim.ts) if you're interested or have any suggestions for it. It's pretty ugly, but as far as I can tell it's accurate. It's pretty clear that #101, #102, and #103 are all gonna be tied closely with this one. My next goal is to add in prestiges so we can run through some scenarios to see how things play out, then we can start fiddling with the actual variables and equations from there. It should be relatively easy to update the script to test out whatever scenarios we want. 

![image](https://user-images.githubusercontent.com/15902166/30008834-3daba01a-90f1-11e7-8b62-cc35bdeaee5e.png)

I can also upload the workbook this chart's in if you want it. I break down # of each generator, what's bought that second, and rate, current total H and overall total H. Oh, and I also only have it purchasing one generator or upgrade per second. I don't imagine that's a limitation that's going to impact anything, but if you think it might, I can update it to handle an arbitrary number per second.

**Comment by angarg12** [comment]
That's absolutely excellent, thanks for your hard work.

You are right, all the balance updates are tightly coupled. I have modified the upgrade system so it is easier to implement new upgrades, which I will probably do soon. The new upgrades will have a big impact on progression so that's something to look forward to.

Once the bot handles prestige it should be much easier to play around with formulas and adjust them.

**Comment by HeinousTugboat** [comment]
Next round of pretty pictures. :-D I got a basic prestige system in place. It doesn't currently *spend* any exotic particles, and it's a pretty naive system, but it works and it does increase generation rate by the power of the exotic particles. So, this paints an even uglier picture than before. You had mentioned that it was better to do numerous short runs, but as far as I can tell, the you should probably make them as short as possible. I've a feeling that prestiging every 1 exotic particle is likely optimal. This kind of makes sense, if you think about it, since the cost of particle n+1 is equal to e^(n+1).

Here's a chart on overall xH gain over a 24 hour period, prestiging when 5, 10, 15, 20, 25, or 50 xH is available:
![image](https://user-images.githubusercontent.com/15902166/30040761-7af832e2-91b1-11e7-956c-07ac16092a73.png)
Here's the corresponding rate graphs for it:
![image](https://user-images.githubusercontent.com/15902166/30040849-65fbb502-91b2-11e7-92a7-ce2e533bf6c3.png)

I have a giant 72mb workbook with all the data stuffed into it if you want. I tried to attach it, but GitHub has a 10mb file size limit. ..that's probably reasonable.

I've added the ability to swap in arbitrary functions for calculating exotic particles, also. Right now it's just doing `(n) => Math.floor(Math.max(0,Math.ln(n))`, same as what's in supernova.js. I can go ahead and plug in some different ones so we can model other curves. Let me know if you have any specific formulae you'd like me to toss in there. I'll also work on adding in the exotic upgrades next. It's starting to get into rougher territory, since we have to model actual strategic decisions.

So, on the note, let me know what you think the optimal **strategy** should be. Do you think one should prestige whenever they reach +50% exotic particle? Whenever they can buy the next exotic upgrade? Whenever they reach 50 more particles? I'm not sure what your goal is for that aspect of the game. Figuring that out can help guide us in balancing the equations towards it. :-)

**Comment by angarg12** [comment]
That's an excellent question and I should definitely clarify that. 

When I think about pacing, I think in terms of real world time instead of number of resources. I divide the game in roughly 4 phases:

* Onboarding: The very early game, essentially up to the first prestige. There is pretty much no decision, only numbers going up. Is the initial hook and introduction to the game. Because it isn't very engaging, it shouldn't last long, tops 1 hour, probably less.
* Early game: The exotic prestige phase with Hydrogen. The game starts to provide choice. You have to choose when to prestige to maximise your gains, and how to spend your prestige currency on upgrades.  Properly balanced it can last 1-5 days.
* Mid-late game: Everything after that when you unlock new elements and new mechanics.
* End game: When you have essentially unlocked all content and are just hunting down the last remaining bits.

The focus of this task is early game. The reason is that currently most players drop the game on this phase. From my current analytics, out of 22 players, 18 make exotic matter, but only 1 gets to unlock new elements. Obviously if players are dropping before reaching the 'good part' of the game, we need to improve their early experience.

Specifically about pacing, I like games that you can properly idle. That means you can check them once in a while (say 1 hour), make a couple of decisions and idle again without feeling that you are missing something. As the game is right now, the optimal way to play is to stay glued to your screen and prestige as fast as possible, and that is not fun. 

So ideally I'm looking for prestige loops that have the sweet spot around 1 hour. Specifically, idling has to pay off. A formula that I found interesting is the following: 

`(log( x)*sqrt(x))/1e2`

This one generates a curve quite similar to what I would expect to work best.

Regarding the bot behaviour buying exotic upgrades, it's ok if it is very simple. The bot can model a 'worst case' where players don't optimise at all, and we expect anyone to do better.

**Comment by HeinousTugboat** [comment]
So, if I can make a suggestion, I would actually rearrange part of the onboarding process. Right now, especially with the big warning about it being in development, it actually feels a lot like Hydrogen is all there is. Obviously messaging can help with that, but I think introducing He and reactions quickly alongside 3H would help realize there's other things to unlock. Essentially combine the Helium exploration with its initial reactions with the Hydrogen prestige phase. The quicker you can get the player to realize that there's a *lot* more than just Hydrogen, I think, the quicker you'll be able to actually hook them.

Also, I just had an idea of combining this issue with #13 by using the actual periodic table and highlighting what's unlocked and how many of each there are. I might put something together along those lines.

> `(log( x)*sqrt(x))/1e2`

Is that Math.log() or Math.ln()?

Finally, are you wanting people to discover elements through reactions or primarily through Dark Matter? I have a feeling I'm probably the one person on your analytics that's unlocked elements, and that's only because I hopped into the console and gave myself a bunch of dark matter just to see what the upgrades were. I had no idea that was how you unlocked them. I can help work on #83 if you like, also. Do you need help with actually implementing the tooltips or just with writing them?

**Comment by angarg12** [comment]
That's Math.log, but it could be in any base. That's one of the parameters to play around with!

The tooltips idea came from a tester in the discord channel. He argues that onboarding is very confusing because players don't understand what each element is or does (e.g. half life). He proposed to add tooltips to explain these nuances. I refuse to put an explicit tutorial that holds your hand explaining everything, so tooltips sound like an ok compromise.

I am all in favours of signaling 'more content to come', such as grayed out tabs or pieces of content, but I am reluctant to introduce new big mechanics earlier. One of the complains of v1 was that mechanics come in too fast and it's confusing. I feel better by pacing them apart and make them upgrades.

In other words, in v1 reactions 'just happen', while in v2 you need to explicitly buy them as an upgrade. That way players will understand better that they have unlocked a new mechanic, since they explicitly bought it. Also, spacing out new mechanics/content give players time to get familiar with them.

Lastly, right now dark matter is the only way to unlock new elements. In the future I would like to introduce new ways to create elements, but for the time being this is the focus. I would like the mid game to feel like a 'sandbox' where you can freely explore the possibilities space of the game e.g. unlock new content and play with it.

Ideally unlocking each element should be hard enough so that choosing an element is meaningful, but not so hard so that if you make a 'poor decision' it sets you back a lot. Being able to unlock about 1 new element a day seems like a reasonable pace, and players could burn through the whole game in about 3 months (I would love if I could keep them engage for so long!).

All being said this is just my current mindset, I am open to changes that make the experience better.

**Comment by HeinousTugboat** [comment]
So, I had to fix my script to work with the latest update to upgrades and generators, but I did get these charts made. Right now I've got it set to prestiging every 30 xH earned or every hour, whichever happens sooner.

![image](https://user-images.githubusercontent.com/15902166/30092710-84770dd8-928e-11e7-9cba-3be0fcdf3ccd.png)

 So, the good news is, `sqrt(x)` definitely does what you want it to. Perhaps too well. With this chart, I just dropped the exotic particle cap and had it prestige every hour. The difference between just `sqrt(x)` and `(log(x)+sqrt(x))/2` is astounding. The progression seems more in line with what you're aiming for, but the numbers are just absurdly huge.

![image](https://user-images.githubusercontent.com/15902166/30092853-6ed436f8-928f-11e7-8989-f2b19d36edcb.png)

The issue, I think, is that `ln(x)` naturally has diminishing returns. I want to point you towards [sigmoid/logistic curves](https://en.wikipedia.org/wiki/Logistic_function). I feel like it would give you far more control over the progression, as you can effectively drop a hard cap on it at any given time, but you could also use it in a diegetic fashion, by suggesting that exotic particles require some kind of containment (hence an upgradeable hard cap), and a reaction speed (catalytic upgrades). [Here](https://www.desmos.com/calculator/n9sixkivt9), I've added a basic logistic curve formula on top of the ones you shared previously, so you can see how we can model it.

![image](https://user-images.githubusercontent.com/15902166/30092421-dba45c7a-928c-11e7-8bbd-2d3a902d2631.png)

Just a thought. Let me know if there are any other sims or aspects you'd like me to run for you. I was drawing up graphs of the rate charts, but they were basically incoherent since the system was prestiging so fast. I'm not real sure what the best way to actually analyze these formulae would be beyond what I'm showing you, so I'd love to hear any ideas you have for where to go next with this piece. If you want to go with the combined or root functions, I can go ahead and run some tests with various constants so we can see how they play out. Also, if you have any other strategies you'd like to see poked at, let me know!

**Comment by angarg12** [comment]
Gotta love those sigmoids!

The shape looks pretty much like what I would like to accomplish. My only issue with it is that in a sigmoid the 'shoulder' happens only once, so progression pretty much stalls after it.

I played around with your formula a bit. First, I multiplied the sigmoid by a sqrt. That way it 'smooths' the curve a bit. I bet this can be done by manipulating the sigmoid parameters properly but my maths wizardry is too weak.

Second, I wanted to fix the problem of having only one 'shoulder', so I came up with the idea that as you get more resources, the sigmoid 'moves' to higher values to produce several shoulders. You can see it in the purple plot [here](https://www.desmos.com/calculator/5ktewqm5j7).

To create a seamless transition I used a sum of the current with previous values. The final formula is extremely ugly but I'm quite happy with the resulting plot. It creates the stacking logs effect that is the pinnacle of idle progression.

![desmos-graph](https://user-images.githubusercontent.com/5389189/30099522-37771eb2-92e6-11e7-8d69-7e380ffcd00c.png)

I would like to smooth out the curve but since production numbers are integers it should work just fine.

If this looks reasonable we could try to plug it into the simulation and see how it affects progression.

**Comment by HeinousTugboat** [comment]
Weeee. This was fun. So. I couldn't match your lines perfectly. At least, not quite. I got pretty darn close though. We could probably tweak it if you're really set on those exact kinds of curves, there's a LOT of little spots we can move around, including a magic number that I'm still pretty puzzled about. Anywho, so I got my semi-continuous version of the formula figured out, plugged it into my sim, then ran my sim with a basic strategy of prestiging every N interval of hours, as long as there is at least 1 exotic particle to be earned, otherwise holding out to earn that one particle.

First, [here](https://www.desmos.com/calculator/dsg9i69sis) is my equation in context of the ones you shared this morning. That magic number, `5.747734128`, is moving the center point of each section of the sigmoid so that the endpoints of each one are essentially continuous. I figured it out by zooming in reeeeeeeeeeeeeeally close and nudging it back and forth until I couldn't see a difference and the numbers were the same on both ends. Importantly, it pretty much only works for the variables we've set right now. I haven't gone through the effort to figure out where it comes from mathematically, so odds are good any adjustments will require a new one.

[Here](https://www.desmos.com/calculator/f6xwt5j5lh)'s my breakdown of how I got to that specific form of the equation, if you're interested. Since the breakpoints essentially just sit where the floor() function flips over to the next order of magnitude, I realized we can.. well.. just add the previous order of magnitude. So that's what it does. Then I jumped through way too many hoops to try and drop out as many of the floor calls as possible. The sims on the naive version of my formula took 13-15 seconds per run, as opposed to 2-3 seconds for all the other ones I've done. While probably not apocalyptic, that's still a huge difference. Luckily, I managed to scrap two of the offending log/floor calls, and I realized we can pretty easily just cache the calculation anyway, since it's basically just finding out how big the number is. Hell, it might even be easier to do it some crazier way like n.toString().length or something. 

Anyway, I think you're right on target with this set of equations! Here's my set of runs, and as far as I can tell, that's pretty much exactly what you were hoping for. The faded line in the background is just a partial plot of the potential xH earned on the 24-hour run. It spikes so hard because the script saves up to buy the higher tier generators. I'm also pretty certain this trend continues over longer timeframes.

![image](https://user-images.githubusercontent.com/15902166/30144905-17969f02-935c-11e7-8f8d-13e725581310.png)

Oh, and I dropped using the root altogether. With sigmoid curves, you can manipulate the exponent on the `e` to smooth or widen the curve out. That's why it's such a gnarly equation. There isn't really any great way to break that bit apart, and the only alternative was to use tanh..which.. I'm not real sure I'm interested in. 

Anywho, there's that. Let me know if you have any other thoughts or would like to see anything. This is probably a good point to start thinking about how we expect the exotic particles to actually be spent.

**Comment by angarg12** [comment]
That looks amazing, excellent job!

I really like how the exotic gains curves look with that formula. For example look at the 6h and 3h curves. Sometimes one is above the other and then they flip, and flip again, so it is not straightforward which strategy is superior. That is the kind of behaviour I want to create.

Right now the formula seems to be too heavily slanted towards idling, but that can be tweaked by parameters later on.

As you say right now it seems more relevant to decide how to spend the exotic matter. I will push soon the two new types of exotic upgrades with placeholder-ish prices. Also we need to rebalance the cost of generators to tune it down. Once we have that in place we can continue simulating and balancing.

**Comment by HeinousTugboat** [comment]
So, I saw your post on /r/incremental_games and wasn't sure whether it'd be better to comment there or here since we've already been talking so heavily about this exact topic, but I figured this would be a good place since there's all this context right here too. 

I was curious to see how the game played right now with semi-optimal play, and I honestly think the biggest issue we're dealing with is simply how slow it is. It's downright painful to get to 100 dark matter. I've been playing more or less actively all day and I'm at all of 27. On top of that, because of diminishing returns on exotic particles and dark matter (`log(log(x))`), it's really best to prestige hard and fast.

To be honest with you, after I bought the x3 production boost for 100 xH, and then prestiged and lost it, I about ragequit. I've been making a list of ideas on how to deal with these issues, but I left most of it at work. I'll definitely share them tomorrow. One big thing, though, would be either reduce the re-purchase cost for production boosts, or make them permanent. As it is, they're essentially pointless while trying to unlock anything.

Another issue I think is that dark matter, the way it's laid out and the fact that it's a second prestige mechanic, comes across as being a very late game thing. Other games have sort of trained us that we don't even bother looking at second prestiges until we've fully explored first prestiges. Since it's essentially impossible to earn eV without spending dark matter, and all of the xH upgrades are purely transient, it feels very misleading. On that aspect, I'd suggest just hiding the eV upgrades until you actually *have* some eV to spend. I'd also really consider either seriously dropping the cost of the table of elements, or even making it free, just so it's more obvious that there's somewhere to go. If you give it a nominal cost of 1 dark matter, then you can use that as a sort of end to the first phase. Right now it feels very end game, and by the time you get to it, it really seems like you've already played the entire game. Hell, I still haven't so much as touched any of the redox mechanics, and the only reactions I've messed with were from cheating and what I remember when I played your original prototype like two years ago. That's why I was originally pushing to get 3H reacting so I could start getting He, I thought that was how it was unlocked originally!

**Comment by angarg12** [comment]
Thanks for the feedback, I really appreciate it.

I have been thinking a lot about this topic for a couple of days. One of the main issues that came to mind from reading people's feedback is how late new mechanics are introduced.

My original idea was that you had to 'work' for the new mechanics. I still like that idea but as it is right now they are introduced too late to keep people engaged.

So I brainstormed a couple of changes to the system. 

First, as you mention, the table won't cost anything. Instead, it will unlock automatically as soon as you generate your first piece of dark matter. The table is new content anyway, not new mechanics, so it should be fine.

Second, and more importantly, new mechanics will be unlocked by spending eV instead of dark matter. This means that players will be able to unlock new mechanics much earlier in the game.

The idea is the following: by introducing new mechanics earlier people will engage more. The catch is that these mechanics start shining once you unlock more elements (particularly reactions).

Still this doesn't address the issue with dark matter. Probably just like log was too gruesome for exotic matter, it is for dark matter. It would make sense to change it to be also a sigmoid.

Another issue as you say is the loss of progress. I am a little bit torn here. Other games manage to get away with people losing their progress, so it is definitely doable. On the other hand I completely understand the frustration.

I am still brainstorming about ways to handle this. On the one hand, there is the whole host of upgrades to speed up the prestige cycle. That could also be applied to dark matter, introduce dark upgrades that boost exotic production. However I would like a more holistic approach where players can play some strategy instead of a simple boost.

That's what I like about things like subatomic particles and the relation between elements. Since you can use these relationships to boost the productions of one another it enables you to strategise the situation.

A couple of ideas on this direction could the following:

* New mechanic: Vault. You have a number of slots where you can favourite upgrades. These upgrades, once bought, will persist through prestiges.
* Vault for elements: The same as vault, but instead of individual upgrades you save whole elements. That way when you make a dark prestige you can still progress.

**Comment by HeinousTugboat** [comment]
So, I spent some time tonight playing with the new prestige function. First and foremost, it's definitely way, way too strong. I was prestiging He before I even earned a single dark matter. The problem is there are way too many synergies and generators are way too cheap. I got to my first exotic prestige in about 10 minutes, where I could've earned about 20 xH. Another 10 minutes and I broke 100.. this was on my very first run. After maybe two or three more cycles over the next 40 minutes, I managed to purchase every single upgrade, had something like 1e9 xH, and by the time I reset for dark matter, I had accrued 4e27 e- and 7.9e31 eV purely from 3H decaying into He.

So, I don't know if that was your goal, but onboarding can *definitely* be completed within an hour now. Pretty handily, truthfully. I have a lot of thoughts about that. So, first, going over your previous post. If we keep a higher turnover rate for exotic particles, I actually think dark matter and losing progress will be fairly OK. Even with something like e30+ H, I only earned somewhere around 150 dark matter. I think if we can corral that feedback loop better, log(log()) will serve us well for that.

I want to take a moment and go back to something I mentioned in passing before but that I think would be a very, very strong solution to our issues. A diegetic interface, if you aren't familiar, is one in a game where the interface elements consist of game elements. I think grounding the game in a more concrete setting may be extremely beneficial here, since we can then build and design upgrades around real world concepts, or some abstraction of them.

So, what does that actually mean? Essentially, that instead of just having Hydrogen generators and Helium generators and Lithium generators and 118 different kinds of generators, we instead of some kind of item that can breed elements, a sort of reactor. We don't need to get into the specifics, suspension of disbelief will carry us plenty far for our purposes here. My idea is that at the start, you have one reactor, and it has Hydrogen in it. It's producing hydrogen, and can only hold so much hydrogen. You can then cycle it out, convert it, or otherwise process it. What that means is there is, at all times, a fundamental hard cap on how much of any particular element you can have. From there, we can add various other upgrades that can increase reaction speeds, conversion speeds, synergy and boosts. The core point here is that we can have upgrades interact with far more than just the speed of production. Once we have a cap and limited progression, we can use upgrades for those too.

There's a second aspect here that I find very useful. We can then introduce a metagame. You start with one reactor. Then you get a second one, and a third, and a fourth. This does two things. It requires strategy, if you can only breed 4 (or however many) elements at any given time, and it drastically simplifies the UI since instead of ultimately trying to juggle 118 different elements all generating simultaneously (which I think might genuinely be impossible to do in anything resembling an optimal fashion...), you only have to focus on your current strategy. In this situation, exotic particles would be generated by whatever flushing or cycling event we can perform to clear out a reactor, and then we can spend some dark matter to re-seed it, or maybe we can use our exotic particles to do it.

See, this is also pretty cleanly going into your ideas regarding the vaults, too. Reactions take place in a reaction chamber. Redox reactions take place in oxidation chambers. Hell, you can even get fancy and start playing with pressures and moles and states if you really want to, but that's probably excessive. The mechanic vault? That's easy, you're just upgrading a reactor. Elemental vault? If we use exotic particles as seeds and dark matter as a currency to buy them among other things, that problem is summarily solved. Of course, the exchange rate has to be offensive at best between them.

To go even deeper, I've been thinking that we could look into adding semi-automation down the road. Say we've maxed out our reactors, and have nowhere to go. What if we can hire researchers or lab assistants or analysts that can automate portions? Say someone that handles a certain reaction elsewhere, freeing up our chamber at the expense of less efficient reactions. Or an analyst that provides us the optimal path to our goal at any given time, but slows down generator production while working.

This also helps deal with an issue I feel like is impending but I haven't been able to really get to yet. Because of the calculations we've been using, like I mentioned in another issue, the most efficient way to move forward is to go wide. The more isotopes and the more elements you can run at a time, the faster you can proceed in the game. The only exception here that I'm aware of is in gathering eV from decay.

Two final thoughts for this particular thread, for now. First, I think we would do well to follow in the footsteps of Kittens and Realm Grinder. Both of them are very long games, with a variety of resources and complex upgrade interplay, that have a very, very strong end game. And both of them use a similar multi-cycle prestige like you're wanting to implement. Second, I think there should be a well defined endgame. Is it purely just to explore everything? If so, then I strongly feel that limiting the number of accessible elements at any given time will be important towards that end. There should be some number of reactors that, given the right combination of elements, will be able to find any result. That will be, I think, the most important place for strategy towards that end.

As always, let me know what I can do to help! Otherwise, I'll keep poking at issues, playtesting and UI work.

**Comment by angarg12** [comment]
Actually I have in my wishlist a feature similar to what you describe.

I call it 'accelerator' as a placerholder. The idea is that you load two source isotopes, smash them together and get a new isotope as a result. For example, load 2H and 2H and get 4He, or 1H and 3H for 4He too. This feature spends eV proportional to the size and depending on the two source isotopes, and it has a yield, so not all isotopes get converted.

You can see hints at this in the game already. Some isotopes have a ratio of 0%. Those isotopes are synthetic, and the only way to unlock them (in the future) is fusion.

To be honest I find your idea of only generating 1 element at a time intriguing to say the least. I like the idea that it reduces the player's cognitive load. But this feature would change almost all the current gameplay, not to mention balancing and side effects. When you swap an element, do all upgrades apply or need to be re bought? Do you buy generators with the current element or with some meta currency?

I already completely reworked gameplay when I went from version 1 to version 2. Also I am cautious about overengineering solutions when the game hasn't been released yet. Right now I would rather get it to a reasonably playable state quickly, release and get feedback. The milestone for the release is balance, bugfixes and UI.

About upgrades and the new prestige loop, I agree that it may be overpowered, but I like the general pace of it. Now it's more a matter of tuning numbers, starting with the generators prices.

BTW the prices that I gave the new upgrades are quickly put together so by all means feel free to play around with them.

**Comment by HeinousTugboat** [comment]
So, I think there was a misunderstanding. I wasn't suggesting any sort of new mechanic, actually. I was just suggesting a way of looking at the setting of the game to help us decide on how to make things work. In fact, the basic ideas can be built in place without changing any major part of what's currently there.

Essentially, instead of just having as many elements available to work on as you have unlocked, add a simple mechanic so that the player has to choose what kinds of generators they can build during a given prestige. Think of how Realm Grinder makes you choose your races. Then, we can add upgrades that allow you to select additional elements.

>  But this feature would change almost all the current gameplay, not to mention balancing and side effects. When you swap an element, do all upgrades apply or need to be re bought? Do you buy generators with the current element or with some meta currency?

Basically, my idea was that you need to purchase a starter generator with exotic particles, and that you could convert dark matter into any exotic particle you've unlocked so far at a steep price. So, as long as I have xH, I can easily build H generators in a slot. If I want to switch to He after I've unlocked it, I would need to either already have xHe, or I would have to convert some dark matter into xHe, then spend that to start the He generation.

> The milestone for the release is balance, bugfixes and UI.

That's actually exactly why I'm suggesting this. Balance and UI with 118 elements active hurts my head.

> Now it's more a matter of tuning numbers, starting with the generators prices.

I actually forgot to mention this, I think. Why's the coefficient for generators so low? 1.05 seems really, really low. Most games are much closer to 1.15, and I think Kittens actually goes up to 1.21 for some things. We should definitely think about increasing that up at least to 1.08 or 1.09. This is another place for potential upgrades, too.

**Comment by angarg12** [comment]
Sorry that I misunderstood your comment!

Comparing it to Realm Grinder and races made it much more clear to me. From that point of view it seems really really interesting.

Would you like to open a new issue to discuss this feature?

**Comment by angarg12** [comment]
Commit f7d036b32e349a5557b04eae598a30e0ad997a9d balances the exotic matter production, dramatically decreasing it.

The new system introduces much stronger diminishing returns. Some breakpoints are:

* 10 EM for 1e8 matter.
* 1000 EM for 1e16 matter.
* 10M EM for 1e42 matter.
* Tops at about 1e24 for 1e303 matter.

**Comment by angarg12** [comment]
@primeprover proposed the following formula for prestige

DM=EXP((-1/2+SQRT(1/4+4*(1/LN(100))*LN(EM/100)))/(2/LN(100)))

I would be sorry to part ways with the sigmoid function. However, we can compare both and see how they work.

**Comment by primeprover** [comment]
This is an adjustment that converts Resources into EM
EM=EXP((-1/2+SQRT(1/4+4*(1/LN(100))*LN(Resources/100000)))/(2/LN(100)))

**Comment by 54czp** [comment]
I already spend 24 hours to get first He atom. I think this game is too difficult to keep playing with few and few positive feedback. I want to know if it is possible to all elements?

**Comment by angarg12** [comment]
Hello, thank you very much for your feedback.

Yes, currently it is possible to unlock all elements. Some elements cannot be bought (they are labelled synthetic) but they can be unlocked by other means.

I know some people who has unlocked all elements way faster than I anticipated.

Do you have any suggestion to make the game more fun during the early part? I think that possibilities increase a lot once you get into dark matter and unlock other elements. If you provide some suggestions I can look into adding them into the game.

**Comment by angarg12** [comment]
By the way, it just occurred to me, are you using the release or the dev version?

I saw a spike of traffic to the game last week for some reason (I don't know why yet). Most people are using the release version.

The dev version is more up to date, and can include balance changes and bugfixes.

https://angarg12.github.io/nucleogenesis-dev/

If you made progress in the release you should be able to import your save.

I'll update the Readme with this information.
